src/topology/algebra/uniform_field.lean

```diff
@@ -0,0 +1,172 @@
+/-
+Copyright (c) 2019 Patrick Massot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Patrick Massot
+-/
+import topology.algebra.uniform_ring
+import topology.algebra.field
+
+/-!
+# Completion of topological fields
+
+The goal of this file is to prove the main part of Proposition 7 of Bourbaki GT III 6.8 :
+
+The completion `hat K` of a Hausdorff topological field is a field if the image under
+the mapping `x ↦ x⁻¹` of every Cauchy filter (with respect to the additive uniform structure)
+which does not have a cluster point at `0` is a Cauchy filter
+(with respect to the additive uniform structure).
+
+Bourbaki does not give any detail here, he refers to the general discussion of extending
+functions defined on a dense subset with values in a complete Hausdorff space. In particular
+the subtlety about clustering at zero is totally left to readers.
+
+Note that the separated completion of a non-separated topological field is the zero ring, hence
+the separation assumption is needed. Indeed the kernel of the completion map is the closure of
+zero which is an ideal. Hence it's either zero (and the field is separated) or the full field,
+which implies one is sent to zero and the completion ring is trivial.
+
+The main definition is `completable_top_field` which packages the assumptions as a Prop-valued
+type class and the main results are the instances `field_completion` and
+`topological_division_ring_completion`.
+-/
+
+
+noncomputable theory
+
+open_locale classical uniformity topological_space
+
+open set uniform_space uniform_space.completion filter
+
+variables (K : Type*) [field K]  [uniform_space K]
+
+local notation `hat` := completion
+
+@[priority 100]
+instance [separated_space K] : nontrivial (hat K) :=
+⟨⟨0, 1, λ h, zero_ne_one $ (uniform_embedding_coe K).inj h⟩⟩
+
+/--
+A topological field is completable if it is separated and the image under
+the mapping x ↦ x⁻¹ of every Cauchy filter (with respect to the additive uniform structure)
+which does not have a cluster point at 0 is a Cauchy filter
+(with respect to the additive uniform structure). This ensures the completion is
+a field.
+-/
+class completable_top_field : Prop :=
+(separated : separated_space K)
+(nice : ∀ F : filter K, cauchy F → 𝓝 0 ⊓ F = ⊥ → cauchy (map (λ x, x⁻¹) F))
+
+local attribute [instance, priority 100] completable_top_field.separated
```

Comment:
Can you add a comment explaining why this instance is local, and why you embed it as a field vs using `extends`?